server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -205,6 +246,65 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
         }
     }
 
+    private long getPoolInterval(Product prod) {
+        long interval = maxCdkLifeDays;
+        String prodThenString = prod.getAttributeValue("expired_after");
+        if (prodThenString != null && Long.parseLong(prodThenString) < maxCdkLifeDays) {
+            interval = Long.parseLong(prodThenString);
+        }
+        return 1000 * 60 * 60 * 24 * interval;
+    }
+
+    private Pool getCdkPool(Consumer consumer, String sku) {
+        List<Pool> pools = poolCurator.listByConsumer(consumer);
+        for (Pool p : pools) {
+            if (p.hasAttribute(Pool.DEVELOPMENT_POOL_ATTRIBUTE) &&
+                p.hasAttribute(Pool.REQUIRES_CONSUMER_ATTRIBUTE) &&
+                p.getAttributeValue(Pool.REQUIRES_CONSUMER_ATTRIBUTE).equals(consumer.getUuid())) {
+                return p;
+            }
+        }
+        return null;
```

Comment:
Why look up all the consumer's pools just to pull out the single dev pool -- seems like wasted resources. Let's do a lookup for that pool via the curator so that we don't have to uselessly pull back all the pool data for the consumer. There should only ever be one dev pool per consumer, right?